src/metrics-logger.ts

```diff
@@ -0,0 +1,77 @@
+'use strict';
+import { Request } from './custom-typings';
+import { Route, RouteSpec, Response, Server } from 'restify';
+import { StatsD, StatsDOptions } from './statsd';
+
+/**
+ * tuple of [seconds, nanoseconds]
+ */
+type HighResolutionTime = [number, number];
+
+export const DEFAULT_KEY_NAME: string = 'handler-0';
+
+export function register(server: Server, statsDOptions: StatsDOptions) {
+    let statsD: StatsD = new StatsD(statsDOptions);
+    let logger = new MetricsLogFactory(statsD).createLogger();
+
+    server.on('after', (request: Request, response: Response, route: Route, error) => {
+        logger(request, response, route);
+    });
+}
+
+export class MetricsLogFactory {
+    constructor(private statsD: StatsD) {
+    }
+
+    createLogger() {
+        return (request: Request, response: Response, route: Route) => {
+            // Ignore requests without timers, e.g. OPTIONS
+            if (!request.timers) {
```

Comment:
This is a great solution for now. I think I would prefer to fall back to using the request time and generating a new date to get the elapsed time, but that should be separate pull request. Please create an issue for this and we'll take care of it later.